wallet/wallet-cli/src/repl/non_interactive/mod.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod log;
+
+use tokio::sync::mpsc;
+
+use crate::{
+    cli_event_loop::Event, cli_println, commands::ConsoleCommand, console::ConsoleContext,
+    errors::WalletCliError,
+};
+
+use super::{get_repl_command, parse_input};
+
+pub fn run<T: std::io::BufRead>(
+    input: std::io::Lines<T>,
+    output: &ConsoleContext,
+    event_tx: mpsc::UnboundedSender<Event>,
+) -> Result<(), WalletCliError> {
+    let repl_command = get_repl_command();
+
+    for line in input {
+        match line {
+            Ok(line) => {
+                let res = parse_input(&line, &repl_command);
+                match res {
+                    Ok(Some(command)) => {
+                        let res = super::run_command_blocking(&event_tx, command);
+
+                        match res {
+                            Ok(cmd) => match cmd {
+                                ConsoleCommand::Print(text) => {
+                                    cli_println!(output, "{}", text);
+                                }
+                                ConsoleCommand::ClearScreen => {
+                                    cli_println!(output, "Not supported");
+                                }
+                                ConsoleCommand::PrintHistory => {
+                                    cli_println!(output, "Not supported");
```

Comment:
When you said you removed history in the meeting, did you mean only for non-interactive mode? Because that's probably OK.

Reply:
No, I meant for interactive mode. History is now not persistent between application runs (unless the history file location is specified on the command line). Non-interactive does not use/need history.

Reply:
But I will change that and make sure that only safe history is stored there.

Reply:
Maybe also see if it's easy to have a shortcut like shift+del to delete a history entry.

Reply:
To be clear, I'm OK with history being persistent just in individual runs. That's acceptable.

Reply:
OK, great. It's done then. And if someone wants persistent history between runs, we have a command-line option for that.